### DISCUSSION TITLE
Problem with motor

### DISCUSSION THREAD

**Discussion by posssson** [doudar/SmartSpin2k#204] [Q&A] [answered]
I have the motor 42SHDC3025-24B and it doesn't work properly. The motor vibrate like the spires aren't corectly wired... Any ideas? I made the good wired, I think.

**Comment by posssson** [comment]
Ok with this

   while (1) {
      for (int x = 0; x < 20000; x++) {
        digitalWrite(STEP_PIN, HIGH);
        delayMicroseconds(2000);
        digitalWrite(STEP_PIN, LOW);
        delayMicroseconds(2000);
      }
      vTaskDelay(100);
    }

the motor turns but very slowly. I don't understand why it doesn't work with a lower value of delayMicrosec...

**Reply by posssson** [reply]
In my case I am in 12V instead of 24V it is the reason I think

**Reply by fireyeti** [reply]
@doudar any ideas? Does the motor turn while not installed on the bike? Double check wiring. My first suggestion is to clean and lubricate the resistance knob of your spin bike, it should be very easy to turn

**Reply by doudar** [reply]
If it vibrates instead of turns, I'd suspect it isn't wired probably between the stepper driver and the motor. 

You'll normally see the wires on the motor side of the cable crossed over like this:


![image](https://user-images.githubusercontent.com/17362216/116796367-6bcd3800-aaa1-11eb-8a81-7eaee5de8cdd.jpeg)

**Comment by posssson** [comment, marked as answer]
The wiring is correct, I found the problem it was the value maxStepperSpeed, my motor doesn't work with 500 but works with 600. Don't know why. But now it is ok.

**Comment by marklogan17** [comment]
Reopening this to try and keep topics together rather than creating a new discussion :)  I ordered the motor in the link below.  When I hooked everything up the software all looked fine, hitting the shifter buttons shows in the log as +/-  600 steps, but the motor didn't move.  Now I'm pretty sure I've killed a couple of stepper drivers, but with the last one carefully all hooked up, I get a little wobble from the motor like its trying to rotate - like it goes one step either way at most.  I've rechecked all the wiring and while my skills are not great, I think its connected.  I actually rebuilt the firmware with the maxStepperSpeed change above - that may or may not have resulted in the minor wobble vs what seemed like no movement before that.  Any other suggestions as to what I could look at?  

https://www.amazon.ca/gp/product/B08C268H2M/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1

**Reply by doudar** [reply]
Thinking out loud:

Make sure your drivers potentiometer is turned up to max. Furthest counterclockwise if it has a stop, otherwise give it about 1/4 turn counter clockwise to start. 

Make sure vmot is connected to a good supply of >12v and the logic supply is connected to 3.3-5v 

Double check step, enable and fit pins are wired correctly.


@fireyeti , anything I'm missing?

**Reply by fireyeti** [reply]
Just double check wiring diagram! I have offset a connector by 1 pin and had interesting results. The voltages are important and check your grounds. Some users have previously reported that certain ground pins on their esp32s were not a good ground. Use a multimeter to confirm. Of note it doesn't matter which ground you are using!

**Reply by marklogan17** [reply]
> Thinking out loud:
> 
> Make sure your drivers potentiometer is turned up to max. Furthest counterclockwise if it has a stop, otherwise give it about 1/4 turn counter clockwise to start.
> 
> Make sure vmot is connected to a good supply of >12v and the logic supply is connected to 3.3-5v
> 
> Double check step, enable and fit pins are wired correctly.
> 
> @fireyeti , anything I'm missing?

I have vmot tied right to the inbound side of the buck converter so its really directly from the power supply.  I have the output of the buck converter exactly at 5.00V.  I've pulled every connection and re-crimped a good percentage of them :)

What haven't I done?  I haven't looked at the driver's potentiometer.  That's a great idea and explains a lot of what I'm seeing.  I'll head back down and check that.

Thank you so much for the ideas.  I'll report back either way.

**Reply by marklogan17** [reply]
> Just double check wiring diagram! I have offset a connector by 1 pin and had interesting results. The voltages are important and check your grounds. Some users have previously reported that certain ground pins on their esp32s were not a good ground. Use a multimeter to confirm. Of note it doesn't matter which ground you are using!

I did pull them all and make sure they're in the right spot.  I found a couple of issues on the stepper driver mostly because I'm too dumb to turn it upside down and put the connections back on the right pins.  I did see mention of the ground pins for the ESPs.  The board seems to be operating correctly and I'm getting input from the shifters correctly so I think the ground is OK but I will do some checking there too.  

Thanks for the ideas.

**Reply by marklogan17** [reply]
I am fairly convinced that through various (different) errors that I have destroyed three separate stepper drivers today.  Luckily I found reasonably cheap replacements that should be here within a couple of days.  While waiting for them I'll re-check all my wiring again.  I'll update when I've tested with the new, known good, stepper drivers.

**Reply by doudar** [reply]
Yikes!

@fireyeti mentioned elsewhere that it might be worthwhile to completely disconnect the wiring between the ESP32 and the stepper driver to start fresh.

I'm sure you noticed also, but keep in mind that the silkscreen on the esp32 is what you should follow, not the physical pin number. So where we say in settings.h "#define STEP_PIN 25", that means GPIO 25 .

**Reply by marklogan17** [reply]
Yes I think I got the pins right based on what is written on the ESP.  Certainly at least I got the shifters right :)  I will definitely pull everything out and be 1000% certain before applying power when the new stepper drivers come in.

**Reply by marklogan17** [reply]
Turns out I only blew up two stepper drivers - third one still works.  I now have all the pieces working.  Now just have to package it :)

**Reply by doudar** [reply]
> Turns out I only blew up two stepper drivers - third one still works. I now have all the pieces working. Now just have to package it :)

Nice!
